src/main/webapp/js/instructorFeedbackResults.js

```diff
@@ -205,40 +205,110 @@ function updateStatsCheckBox() {
+ */
+function expandPanels(element, panels) {
+    // Expand/collapse buttons on AJAX-loaded panels have a collapse-panels-button class
+    var areAjaxLoadedPanels = $(element).is($('#collapse-panels-button'));
+    var TWENTIETH_SECOND_TO_MILLI_SECOND = 50;
+
+    for (var idx = 0; idx < panels.length; idx++) {
+        expandPanel(panels[idx], idx * TWENTIETH_SECOND_TO_MILLI_SECOND, areAjaxLoadedPanels);
+    }
+}
+
+/**
+ * Expands {@code panel}. If the panel data is not loaded, load it by Ajax first before expansion.
+ * @param {int} timeOut - is in milliseconds
+ */
+function expandPanel(panel, timeOut, isAjaxLoadedPanel) {
+    var isPanelAlreadyExpanded = $(panel).hasClass('in');
+    if (isPanelAlreadyExpanded) {
+        return;
+    }
+
+    if (isAjaxLoadedPanel) {
+        var isElementClickedToLoadAjax = checkAndLoadPanelByAjax(panel);
+        if (isElementClickedToLoadAjax) {
+            // When the element has Ajax class, its panel is in collapsed state. Clicking will expand
+            // its panels.
+            return;
+        }
+    }
+    // expand this panel
+    setTimeout(showSingleCollapse, timeOut, panel);
+}
+
+/**
+ * Checks whether the panel data is loaded. If not, load panel data by Ajax.
+ * @return true if Ajax loading happened
+ */
+function checkAndLoadPanelByAjax(panel) {
+    // When the panel's parent element has class ajax_auto or ajax-response-auto(not both), the panel data
+    // has not been loaded yet. We need to load the panel data by Ajax.
+    // ajax_auto class is for loading normal panel data while ajax-response-auto is for loading users
+    // who have not responded.
+    var $ajaxElement = $(panel).parent().children('.ajax_auto');
+    if ($ajaxElement.length === 0) {
+        $ajaxElement = $(panel).parent().children('.ajax-response-auto');
+    }
+
+    var hasAjaxElement = $ajaxElement.length !== 0;
+    if (hasAjaxElement) {
+        // Trigger Ajax loading by clicking on the element. When clicked, the panel data is loaded,
+        // and ajax_auto or ajax-response-auto class will be removed from the element.
+        $ajaxElement.click();
+    }
+    return hasAjaxElement;
+}
+
+function collapsePanels(panels) {
+    var TENTH_SECOND_TO_MILLI_SECOND = 100;
+    for (var idx = 0; idx < panels.length; idx++) {
+        var isPanelAlreadyCollapsed = !$(panels[idx]).hasClass('in');
+        if (isPanelAlreadyCollapsed) {
+            continue;
         }
-        var htmlStr = $(e).html();
-        htmlStr = htmlStr.replace(collapse, expand);
-        $(e).html(htmlStr);
-        var tooltipStr = $(e).attr('data-original-title').replace(collapse, expand);
-        $(e).attr('title', tooltipStr).tooltip('fixTitle').tooltip('show');
+        // collapse this panel
+        setTimeout(hideSingleCollapse, idx * TENTH_SECOND_TO_MILLI_SECOND, panels[idx]);
     }
 }
 
+/**
+ * Replace the Html content after the panel expands/collapses.
```

Comment:
What html content exactly? Need to explain more, for example, the fact that the tooltip will be updated.
Is this function for panels only? The name doesn't suggest any such restriction but the header comment does.